tests/integration/test_sampling.py

```diff
@@ -368,3 +368,120 @@ def test_rate_limiter_on_long_running_spans(tracer):
+                pass
+        with tracer.trace("donkey_monkey") as span4:
+            pass
+
+    # Trace level sampling decision tags should be the same.
+    for span in [root, span2, span3, span4]:
+        assert span.context.sampling_priority == -1, repr(span)
+        assert span.context._meta.get("_dd.p.dm") == "-3", repr(span)
+
+    # Span 1 was sampled via trace sampling rule
+    assert root.get_metric("_dd.rule_psr") == 0, repr(root)
+    # Span 2 was sampled via single span sampling rule
+    assert span2.get_metric("_dd.span_sampling.mechanism") == 8, repr(span2)
+    assert span2.get_metric("_dd.span_sampling.rule_rate") == 1, repr(span2)
+    assert "_dd.rule_psr" not in span2.get_metrics(), repr(span2)
+
+
+@pytest.mark.skipif(AGENT_VERSION != "testagent", reason="Tests only compatible with a testagent")
+@pytest.mark.snapshot()
+@pytest.mark.subprocess(
+    env={
+        "DD_SERVICE": "animals",
+        "DD_TRACE_PARTIAL_FLUSH_MIN_SPANS": "1",
+        "_DD_TRACE_WRITER_NATIVE": "false",
+        "DD_TRACE_SAMPLING_RULES": '[{"sample_rate": 0, "service": "animals"}]',
+        "DD_SPAN_SAMPLING_RULES": '[{"service":"animals", "name":"monkey", "sample_rate":1}]',
+    },
+    parametrize={"DD_TRACE_COMPUTE_STATS": ["false", "true"]},
+)
+def test_single_span_and_trace_sampling_match_root_span_paritial_flushing():
+    """Validates that a single span sampling rule applied to a root span does not
+    override the trace sampling decision, even when traces are partially flushed.
+    """
+    from ddtrace.trace import tracer
+
+    with tracer.trace("monkey") as root:
+        pass
+    with tracer.start_span("non_monkey", child_of=root) as span2:
+        pass
+
+    for span in [root, span2]:
+        # Trace sampling rule is matched, trace level sampling decision is MANUAL DROP
+        assert span.context.sampling_priority == -1, repr(span)
+        assert span.context._meta.get("_dd.p.dm") == "-3", repr(span)
+
+    # Span 1 was sampled via single span sampling rule
+    assert root.get_metric("_dd.span_sampling.mechanism") == 8
+    assert root.get_metric("_dd.span_sampling.rule_rate") == 1
+    # Regardless of whether stats computation is enabled or disabled,
+    # if the trace matched a trace sampling rule `_dd.rule_psr` should be set.
+    assert root.get_metric("_dd.rule_psr") == 0, repr(root)
+
+
+@pytest.mark.skipif(
+    AGENT_VERSION != "testagent" or USING_NATIVE_WRITER is False,
+    reason="Tests only compatible with a testagent and native writer",
+)
+@pytest.mark.subprocess(
+    env={
+        "DD_SERVICE": "animals",
+        "DD_TRACE_SAMPLING_RULES": '[{"sample_rate": 0, "service": "animals"}]',
+        "DD_SPAN_SAMPLING_RULES": '[{"service":"animals", "name":"monkey", "sample_rate":1}]',
+    },
+    parametrize={"DD_TRACE_COMPUTE_STATS": ["true", "false"]},
+)
+@pytest.mark.snapshot()
+def test_single_span_and_trace_sampling_match_paritial_flushing_native_writer():
```

Comment:
is partial flushing happening here?

Reply:
Actually I will remove this test and add `@pytest.mark.xfail(USING_NATIVE_WRITER, reason="Native writer does not drop traces when partial flushing is enabled")` to the other two tests. 

I don't think having a test case just for the native writer makes sense. In a follow up PR we should ensure the native writer and the agent writer send the same traces when stats computation is enabled.